src/snowflake/cli/plugins/nativeapp/teardown_processor.py

```diff
@@ -226,8 +226,13 @@ def drop_package(self, auto_yes: bool):
             )
             if show_versions_cursor.rowcount is None:
                 raise SnowflakeSQLExecutionError(show_versions_query)
+
             if show_versions_cursor.rowcount > 0:
-                raise CouldNotDropApplicationPackageWithVersions()
+                # allow dropping a package with versions when --force is set
+                if not auto_yes:
+                    raise CouldNotDropApplicationPackageWithVersions(
+                        "Drop versions first, or use --force to override."
```

Comment:
what happens to the versions associated with the app package in that case?

Reply:
They are obliterated along with the package (they are essentially sub-objects).